rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs

```diff
@@ -59,3 +63,96 @@ fn test_create_neuron() {
         }));
     });
 }
+
+#[test]
+fn test_set_neuron_dissolve_delay_timestamp() {
+    let rt = Runtime::new().unwrap();
+    rt.block_on(async {
+        let env = RosettaTestingEnvironment::builder()
+            .with_initial_balances(
+                vec![(
+                    AccountIdentifier::from(TEST_IDENTITY.sender().unwrap()),
+                    // A hundred million ICP should be enough
+                    icp_ledger::Tokens::from_tokens(100_000_000).unwrap(),
+                )]
+                .into_iter()
+                .collect(),
+            )
+            .with_governance_canister()
+            .build()
+            .await;
+
+        // Stake the minimum amount 100 million e8s
+        let staked_amount = 100_000_000u64;
+        let neuron_index = 0;
+        let from_subaccount = [0; 32];
+
+        env.rosetta_client
+            .create_neuron(
+                env.network_identifier.clone(),
+                &(*TEST_IDENTITY).clone(),
+                RosettaCreateNeuronArgs::builder(staked_amount.into())
+                    .with_from_subaccount(from_subaccount)
+                    .with_neuron_index(neuron_index)
+                    .build(),
+            )
+            .await
+            .unwrap();
+
+        // See if the neuron was created successfully
+        let agent = get_test_agent(env.pocket_ic.url().unwrap().port().unwrap()).await;
+        let neuron = list_neurons(&agent).await.full_neurons[0].to_owned();
+
+        let dissolve_delay_timestamp = match neuron.dissolve_state.unwrap() {
+            // When a neuron is created its dissolve delay timestamp is set to now which corresponds to the state DISSOLVED
+            DissolveState::WhenDissolvedTimestampSeconds(dissolve_delay_timestamp) => {
+                dissolve_delay_timestamp
+            }
+            k => panic!(
+                "Neuron should be in WhenDissolvedTimestampSeconds state, but is instead: {:?}",
+                k
+            ),
+        };
+        assert_eq!(
+            dissolve_delay_timestamp,
+            env.pocket_ic
+                .get_time()
+                .await
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_secs()
+        );
```

Comment:
I think the state machine time only ticks forward if you explicitly tell it to no?

Reply:
Not in this case, since the `pocket_ic` instance in the `RosettaTestingEnvironment` is set to live mode [here](https://sourcegraph.com/github.com/dfinity/ic@4bd8e1d8430ee17c7e3be47732d1786074ecb592/-/blob/rs/rosetta-api/icp/tests/system_tests/common/system_test_environment.rs?L321&symbol=rust-analyzer+cargo+ic-rosetta-api+2.1.0+system_test_environment/common/RosettaTestingEnvironmentBuilder%23build%28%29.).

Reply:
I see, thanks for catching this. I'll make the change.